client/main/editors/transformeditor.js

```diff
@@ -0,0 +1,483 @@
+'use strict';
+
+const $ = require('jquery');
+const keyboardJS = require('keyboardjs');
+const opsDropDown = require('./operatordropdown');
+const tarp = require('../utils/tarp');
+const formulaToolbar = require('../vareditor/formulatoolbar');
+const dropdown = require('../vareditor/dropdown');
+
+const TransformEditor = function(dataset) {
```

Comment:
so in this 'class' you're co-mingling what would normally be the constructor, and the method definitions. do you want to adopt the `class` design?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

Reply:
didn't even know that js had classes